src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationServiceTest.kt

```diff
@@ -482,6 +486,38 @@ internal class RecommendationServiceTest : ServiceTestBase() {
     }
   }
 
+  @Test
+  fun `given invalid recall type should throw invalid request exception`() {
+    runTest {
+      val existingRecommendation = RecommendationEntity(
+        id = 1,
+        data = RecommendationModel(
+          crn = crn,
+          recallType = RecallType(
+            selected = RecallTypeSelectedValue(RecallTypeValue.NO_RECALL),
+            allOptions = listOf(TextValueOption("NO_RECALL"))
+          )
+        )
+      )
+      val updateRecommendationRequest = RecommendationModel(
+        crn = crn,
+        recallType = existingRecommendation.data.recallType?.copy(
+          selected = RecallTypeSelectedValue(RecallTypeValue.FIXED_TERM),
+          allOptions = listOf(TextValueOption("NO_RECALL"))
+        )
+      )
+      val json = CustomMapper.writeValueAsString(updateRecommendationRequest)
+      val recommendationJsonNode: JsonNode = CustomMapper.readTree(json)
+
+      try {
+        recommendationService.updateRecommendation(recommendationJsonNode, 1L, "Bill", null, false)
+      } catch (e: InvalidRequestException) {
```

Comment:
How does the test know if the exception has been thrown? Do you not need an assertion somewhere?

Reply:
It's implied. The following line asserts that the recommendation repo has no further interactions. If the exception was never thrown, the update would proceed and the test would fail as an interaction would happen. We have used the same test pattern elsewhere. The controller also tests the exception is thrown by checking that a 400 status is returned on an invalid recall type.

Reply:
ok